smarts/core/smarts.py

```diff
@@ -898,9 +898,15 @@ def _pybullet_provider_sync(self, provider_state: ProviderState):
                         vehicle_id=vehicle_id,
                         vehicle_config_type=vehicle.vehicle_config_type,
                     )
-                # Update the social vehicle avatar to match the vehicle state
+
+                # Update social vehicle pose when no active agents are present
+                if not self._vehicle_index.agent_vehicle_ids():
```

Comment:
I could suggest calling `agent_vehicle_ids()` once outside the loop since there is a significant cost to the method.

Reply:
Moved `agent_vehicle_ids` to outside `for` loop.